game-core/src/main/java/games/strategy/engine/framework/ServerGame.java

```diff
@@ -442,22 +442,23 @@ private void runStep(final boolean stepIsRestoredFromSavedGame) {
     if (gameData.getSequence().next()) {
       gameData.getHistory().getHistoryWriter().startNextRound(gameData.getSequence().getRound());
       saveGame(gameData.getSequence().getRound() % 2 == 0
-          ? AutoSaveFileUtils.getEvenRoundAutoSaveFile(headless)
-          : AutoSaveFileUtils.getOddRoundAutoSaveFile(headless));
+          ? launchAction.getAutoSaveFileUtils().getEvenRoundAutoSaveFile()
+          : launchAction.getAutoSaveFileUtils().getOddRoundAutoSaveFile());
     }
     if (autoSaveThisDelegate && !currentStep.getName().endsWith("Move")) {
-      autoSaveAfter(currentDelegate, headless);
+      autoSaveAfter(currentDelegate);
     }
   }
 
-  private void autoSaveAfter(final String stepName, final boolean headless) {
-    saveGame(AutoSaveFileUtils.getAfterStepAutoSaveFile(stepName, headless));
+  private void autoSaveAfter(final String stepName) {
+    final var saveUtils = launchAction.getAutoSaveFileUtils();
+    saveGame(saveUtils.getAfterStepAutoSaveFile(saveUtils.getAutoSaveStepName(stepName)));
```

Comment:
I think I see a code smell here where an object is called to compute a value, then that value is passed to another method in the same object. EG: `a.bar(a.foo(value))`, can be converted to `a.bar(value)`, ie: `saveUtils.getAfterStepAutoSaveFile(stepName)`